### ISSUE TITLE
Doesn't it make more sense to have a colorscale option enabled for the scatterplot?

### ISSUE THREAD

**Issue by vidyap-xgboost** [jbesomi/texthero#98]
According to [Plotly's documentation](https://plotly.com/python/colorscales/), the `colorscale` for any plot is automatically assigned depending on whether the given data is numeric or categorical. In hero.scatterplot() function, the argument `color` is given as `kmeans` and `hover_data` is usually given as `clean_text` to make the user understand which data point belongs to which text.

I feel it makes more sense to have a `colorscale` argument where we can give custom colours (mostly to enable colours which are given to categorical variables) so that the distinction is clear between data points.

Consider the below scatterplot where I have 80k+ data points and they're divided into 7 clusters:

<img width="955" alt="2020-07-16 (2)" src="https://user-images.githubusercontent.com/65642130/87632133-5ecf9a00-c756-11ea-94c3-3984f632b63d.png">

I would want to separate the `kmeans` in distinct colours rather than `continuous` colours because I want to see which data point belongs to which `clean_text` (especially when the `dataset` is huge!). When there are continuous colours it makes it harder to have a clear distinction between different `clusters`.

If I'm not making sense or if am wrong here, let's discuss!

**Comment by jbesomi** [comment]
Hey @vidyap-xgboost, good you are pointing this out, 100% agree and was on the "todo" list.

Solved in a93cc06: "clustering" functions now returns a "category" Pandas Series.

Other comments:

1. Adding the `colorscale` argument might not be strictly necessary. The idea is to keep this function very simple; otherwise the users can write its own function directly using Plotly (that's just an opinion, as we know the `visualization` module is very opinionable ...)
2. We shouldn't set the type to the `color` argument into the `scatterplot` function as we can simply respect the Plotly rules:
   1. If `color` is a Pandas Series with digit values -> continuous color bar
   2. Otherwise if `color` is an `object` (str) or a `category` Pandas Series -> discrete color bar

What's next:

1. I realized that the three "clustering" functions do not have a complete docstring. Do you mind open an `issue` with label `documentation` and say we should improve it? This will be similar to #62 (as idea)
2. We should update the docstring of `scatterplot` and explain what we discussed related to the `color` argument. Again, we should open an issue (or if you wish you can  directly add a PR)

Thank you again! 🎉

**Comment by vidyap-xgboost** [comment]
> Hey @vidyap-xgboost, good you are pointing this out, 100% agree and was on the "todo" list.
> 
> Solved in [a93cc06](https://github.com/jbesomi/texthero/commit/a93cc067e3aa6b80621d828fe7a13a5712cb8160): "clustering" functions now returns a "category" Pandas Series.
> 
> Other comments:
> 
> 1. Adding the `colorscale` argument might not be strictly necessary. The idea is to keep this function very simple; otherwise the users can write its own function directly using Plotly (that's just an opinion, as we know the `visualization` module is very opinionable ...)
> 2. We shouldn't set the type to the `color` argument into the `scatterplot` function as we can simply respect the Plotly rules:
>    
>    1. If `color` is a Pandas Series with digit values -> continuous color bar
>    2. Otherwise if `color` is an `object` (str) or a `category` Pandas Series -> discrete color bar
> 
> What's next:
> 
> 1. I realized that the three "clustering" functions do not have a complete docstring. Do you mind open an `issue` with label `documentation` and say we should improve it? This will be similar to #62 (as idea)
> 2. We should update the docstring of `scatterplot` and explain what we discussed related to the `color` argument. Again, we should open an issue (or if you wish you can  directly add a PR)
> 
> Thank you again! 🎉

I am glad you worked on returning a `category`. I see that it was recently merged and haven't kept track of that PR these past few days, sorry for that. I will test it once I get time on my hands. Thanks for keeping me up to date.

I totally agree with the 2 points you mentioned about `colorscale` argument, makes a lot of sense. 👍 

I better create 2 separate issues for docstring for all the 3 "clustering" functions and `scatterplot` to keep a track of them.

I'm not sure if I have the permissions to add **label** to issues when I create them. I don't see that **settings** icon. Is there another way?

**Comment by jbesomi** [comment]
> I'm not sure if I have the permissions to add labels to issues when I create them. I don't see that settings icon. Is there another way?

Will look into it more deeply later, as far as I know, you should be able to add this for issues you created if you are a contributor (and you are!), isn't? (can you see the settings icon?)

<img width="323" alt="Screenshot 2020-07-16 at 13 14 18" src="https://user-images.githubusercontent.com/43236409/87664887-51221080-c766-11ea-91c9-d2fe5e093ac2.png">

> I better create 2 separate issues for docstring for all the 3 "clustering" functions and scatterplot to keep a track of them.

Perfect!

> I am glad you worked on returning a category. I see that it was recently merged and haven't kept track of that PR these past few days, sorry for that. I will test it once I get time on my hands. Thanks for keeping me up to date.

No problem at all! Also, I just added that about 30 minutes ago .. :)

**Comment by vidyap-xgboost** [comment]
>> Will look into it more deeply later, as far as I know, you should be able to add this for issues you created if you are a contributor (and you are!), isn't? (can you see the settings icon?)

Yes, I have `contributor` badge but no permission to add or change labels and no settings icon is visible.
<img width="377" alt="Annotation 2020-07-16 165558" src="https://user-images.githubusercontent.com/65642130/87665986-75d9b080-c785-11ea-9e0a-1039c627b5aa.png">
